scripts/zones/Dynamis-Xarcabard/mobs/Vanguard_Eye.lua

```diff
@@ -35,27 +35,47 @@ end;
 -----------------------------------
 
 function onMobDeath(mob,killer)
-	
-	local mobID = mob:getID();
-	
-	-- 035 039: spawn 043 when defeated
-	if (mobID == 17326536) then
-		SpawnMob(17326553);
-	-- 058: spawn 60 when defeated
-	elseif (mobID == 17326661 and GetMobAction(17326668) == 0 and GetMobAction(17326673) == 0 or 
-		   mobID == 17326668 and GetMobAction(17326661) == 0 and GetMobAction(17326673) == 0 or 
-		   mobID == 17326673 and GetMobAction(17326661) == 0 and GetMobAction(17326668) == 0) then
-		SpawnMob(17326706);
-	-- 114: spawn 112 when defeated
-	elseif (mobID == 17326790) then
-		SpawnMob(17326086);
-	-- 144-149: spawn 150 when defeated
-	elseif (mobID >= 17330913 and mobID <= 17330918) then
-		SetServerVariable("[DynaXarcabard]TE150_Trigger",GetServerVariable("[DynaXarcabard]TE150_Trigger") + (mobID - 17330912) ^ 2);
-	end
-	
-	if (GetServerVariable("[DynaXarcabard]TE150_Trigger") == 63) then
-		SpawnMob(17330919); -- 150
-	end
-	
-end;
+    
+    local mobID = mob:getID();
+    
+    -- 035 039: spawn 043 when defeated
+    if (mobID == 17330718) then
+        local southTE = GetServerVariable("[DynaXarcabard]TE43_Trigger");
+        if (bit.band(southTE, bit.lshift(1,0)) ~= 1) then
+            SetServerVariable("[DynaXarcabard]TE43_Trigger",bit.bor(southTE, bit.lshift(1,0)))
+        end
+    elseif (mobID == 17330756)  then
+        local southTE = GetServerVariable("[DynaXarcabard]TE43_Trigger");
+        if (bit.band(southTE, bit.lshift(1,1)) ~= 1) then
+            SetServerVariable("[DynaXarcabard]TE43_Trigger",bit.bor(southTE, bit.lshift(1,1)))
+        end
+    -- 058: spawn 60 when defeated
+    elseif (mobID >= 17330827 and mobID <= 17330829) then
+        local northTE = GetServerVariable("[DynaXarcabard]TE60_Trigger");
+        if (bit.band(northTE, bit.lshift(1, (mobID - 17330827))) ~= 1) then
+            SetServerVariable("[DynaXarcabard]TE60_Trigger",bit.bor(northTE, bit.lshift(1, (mobID - 17330827))))
+        end
+    -- 114: spawn 112 when defeated
+    elseif (mobID == 17326790) then
+        SpawnMob(17326086);
+    -- 144-149: spawn 150 when defeated
+    elseif (mobID >= 17330913 and mobID <= 17330918) then
+        local wallTE = GetServerVariable("[DynaXarcabard]TE150_Trigger");
+        if (bit.band(wallTE, bit.lshift(1, (mobID - 17330913))) ~= 1) then
+            SetServerVariable("[DynaXarcabard]TE150_Trigger",bit.bor(wallTE, bit.lshift(1, (mobID - 17330913))))
+        end
+    end
+    
+    if (GetServerVariable("[DynaXarcabard]TE150_Trigger") == 63) then

```

Comment:
don't you think it's a bit silly to get the server variable, flip a bit, save it, and then load it again to check to spawn? and why check the other two server variables if you know only one can ever be checked once (the one that got its bit flipped)?